cluster/schema/manager.go

```diff
@@ -38,9 +39,9 @@ type SchemaManager struct {
 	log    *logrus.Logger
 }
 
-func NewSchemaManager(nodeId string, db Indexer, parser Parser, log *logrus.Logger) *SchemaManager {
+func NewSchemaManager(nodeId string, db Indexer, parser Parser, classTenantDataEvents chan metadataserver.ClassTenant, log *logrus.Logger) *SchemaManager {
```

Comment:
I would prefer to create another constructor like `NewSchemaManagerWithTenantEvents` (something like that) and use only that on our `exp/query` path.

That way we don't touch existing schemaManager and it's very explicit. Plus when not running our `exp/querier` people don't have to worry about this `classTenantDataEvents` at all. 

Wdyt? Happy to know if I'm missing any of your rationale for doing this way.